lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -21,6 +22,66 @@ const meta = {
 	fixable: true,
 };
 
+/** @type {(node: import('css-tree').CssNode) => node is import('css-tree').CssNode & { children: List<import('css-tree').CssNode> }} */
+const hasChildren = (node) => 'children' in node && node.children instanceof List;
+
+/** @type {(node1: import('css-tree').CssNode[], node2: import('css-tree').CssNode[]) => boolean} */
```

Comment:
[suggest] For readability, how about adding a type alias like this?

```suggestion
/** @typedef {import('css-tree').CssNode} CssNode */

/** @type {(node: CssNode) => node is CssNode & { children: List<CssNode> }} */
const hasChildren = (node) => 'children' in node && node.children instanceof List;

/** @type {(node1: CssNode[], node2: CssNode[]) => boolean} */
```

Reply:
Thanks! I didn't know `@typedef`.
Fixed to use `@typedef`. https://github.com/stylelint/stylelint/pull/6772/commits/b81cc2335b4e932c73a11ffde9f040b553e403e6